src/os/user/iterate_cgo_bsd_test.go

```diff
@@ -0,0 +1,43 @@
+//go:build (darwin || freebsd || openbsd || netbsd) && cgo && !osusergo
+// +build darwin freebsd openbsd netbsd
+// +build cgo
+// +build !osusergo
+
+package user
+
+import (
+	"errors"
+	"testing"
+)
+
+// As cgo implementation can not be tested on darwin or bsd based operating
+// systems, since there is no support for fgetpwent/fgetgrent library calls,
+// we will attempt to check if user/group record could be at least retrieved
```

Comment:
As BSDs (including darwin) do not support fgetpwent/fgetgrent, attempt to check if at least 1 user/group record can be retrieved.